test/index.spec.js

```diff
@@ -93,81 +83,79 @@ describe('datastore-pubsub', function () {
     ++testCounter
-    done()
   })
 
-  after(function (done) {
-    parallel([
-      (cb) => stopDaemon(ipfsdA, cb),
-      (cb) => stopDaemon(ipfsdB, cb)
-    ], done)
+  after(() => {
+    return Promise.all([
+      stopDaemon(ipfsdA),
+      stopDaemon(ipfsdB)
+    ])
   })
 
-  it('should subscribe the topic, but receive error as no entry is stored locally', function (done) {
+  it('should subscribe the topic, but receive error as no entry is stored locally', async () => {
     const dsPubsubA = new DatastorePubsub(pubsubA, datastoreA, peerIdA, smoothValidator)
     const subsTopic = keyToTopic(`/${keyRef}`)
 
-    pubsubA.ls((err, res) => {
-      expect(err).to.not.exist()
-      expect(res).to.exist()
-      expect(res).to.not.include(subsTopic) // not subscribed key reference yet
-
-      dsPubsubA.get(key, (err) => {
-        expect(err).to.exist() // not locally stored record
-        expect(err.code).to.equal('ERR_NOT_FOUND')
-
-        pubsubA.ls((err, res) => {
-          expect(err).to.not.exist()
-          expect(res).to.exist()
-          expect(res).to.include(subsTopic) // subscribed key reference
-          done()
-        })
-      })
-    })
+    let subscribers = await pubsubA.ls()
+
+    expect(subscribers).to.exist()
+    expect(subscribers).to.not.include(subsTopic) // not subscribed key reference yet
+
+    let res
+    try {
+      res = await dsPubsubA.get(key)
+    } catch (err) {
+      expect(err).to.exist() // not locally stored record
+      expect(err.code).to.equal('ERR_NOT_FOUND')
+    }
+    expect(res).to.not.exist()
+
+    subscribers = await pubsubA.ls()
+
+    expect(subscribers).to.exist()
+    expect(subscribers).to.include(subsTopic) // subscribed key reference
   })
 
-  it('should put correctly to daemon A and daemon B should not receive it without subscribing', function (done) {
+  it('should put correctly to daemon A and daemon B should not receive it without subscribing', async () => {
     const dsPubsubA = new DatastorePubsub(pubsubA, datastoreA, peerIdA, smoothValidator)
     const dsPubsubB = new DatastorePubsub(pubsubB, datastoreB, peerIdB, smoothValidator)
     const subsTopic = keyToTopic(`/${keyRef}`)
 
-    pubsubB.ls((err, res) => {
-      expect(err).to.not.exist()
-      expect(res).to.exist()
-      expect(res).to.not.include(subsTopic) // not subscribed
-
-      dsPubsubA.put(key, serializedRecord, (err) => {
-        expect(err).to.not.exist()
-        dsPubsubB.get(key, (err, res) => {
-          expect(err).to.exist() // did not receive record
-          expect(res).to.not.exist()
-          done()
-        })
-      })
-    })
+    const res = await pubsubB.ls()
+    expect(res).to.exist()
+    expect(res).to.not.include(subsTopic) // not subscribed
+
+    await dsPubsubA.put(key, serializedRecord)
+
+    try {
+      await dsPubsubB.get(key)
+      expect.fail('Should not have received message')
+    } catch (err) {
+      expect(err.code).to.equal('ERR_NOT_FOUND')
+    }
```

Comment:
Same as above, use `.then(expect.fail, (err) => {})`. While this won't produce false positives, it's consistent with the above and should help prevent updating tests to include potential false positives in the future.

Reply:
There are a lot of these in the tests. I would do a search for `try` in the tests to find all of them and replace with .then logic. There really shouldn't be any try/catch logic in tests, imo.

Reply:
> There really shouldn't be any try/catch logic in tests, imo

I agree, but the design of this module is to use exceptions as flow control, e.g. getting a given topic key subscribes you to that topic as a side effect.

This should be changed but is outside the scope of this PR.